test/e2e/catsrc_pod_config_e2e_test.go

```diff
@@ -0,0 +1,200 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8scontrollerclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const catalogSourceLabel = "olm.catalogSource"
+
+var _ = By
+
+var _ = Describe("CatalogSource Grpc Pod Config", func() {
+	var (
+		generatedNamespace corev1.Namespace
+	)
+
+	BeforeEach(func() {
+		generatedNamespace = SetupGeneratedTestNamespace(genName("catsrc-grpc-pod-config-e2e-"))
+	})
+
+	AfterEach(func() {
+		TeardownNamespace(generatedNamespace.GetName())
+	})
+
+	When("the user wants more control over where the grpc catalog source pod gets scheduled", func() {
+		var (
+			client              k8scontrollerclient.Client
+			catalogSource       *v1alpha1.CatalogSource
+			defaultNodeSelector = map[string]string{
+				"kubernetes.io/os": "linux",
+			}
+			defaultTolerations       []corev1.Toleration = nil
+			catalogSourceName                            = "test-catsrc"
+			defaultPriorityClassName                     = ""
+		)
+
+		BeforeEach(func() {
+			client = ctx.Ctx().Client()
+
+			// must be a grpc source type with spec.image defined
+			catalogSource = &v1alpha1.CatalogSource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      catalogSourceName,
+					Namespace: generatedNamespace.GetName(),
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      "repo/image:tag",
+				},
+			}
+		})
+
+		AfterEach(func() {
+			// assume the catalog source was created and just delete it
+			_ = client.Delete(context.TODO(), catalogSource)
+
+			// wait for it to go away
+			Eventually(func() bool {
+				err := client.Get(context.TODO(), k8scontrollerclient.ObjectKey{
+					Name:      catalogSource.GetName(),
+					Namespace: catalogSource.GetNamespace(),
+				}, &v1alpha1.CatalogSource{})
+				return k8serror.IsNotFound(err)
+			}).Should(BeTrue())
```

Comment:
Another non-blocking nit: I'll be surprised if this isn't already there in a `utils`, but this looks like code that'll be repeated fairly often. Something along the line of `deleteObj(client, name, namespace)` func could help reduce the redundancy.

Reply:
This is a great point - I will check if I can find anything